csrc/device_lower/pass/grid_serialization.cpp

```diff
@@ -127,7 +127,8 @@ class GridSerializationSyncInserter : kir::ExprMutator {
     kir::Allocate* alloc = lower_utils::allocGlobalBufferForGridComm(
         lower_utils::getGridSyncBufferSize(cur_expr_sync_pattern_.value()),
         DataType::Int,
-        true);
+        /*zero_init=*/true,
+        /*resets_to_zero=*/true);
```

Comment:
This is the idiomatic way to create a semaphore buffer that will be reused. This should only be done if we can guarantee that the buffer is clean. Note that for debugging you can enable `NVFUSER_DUMP=global_zeroed_memory` to test whether the semaphores are actually cleaned, as that will count non-zero bytes in the global zeroed memory pool.